From 14edcf5dd0c554a069ac56b049f8eb8e941d2a3e Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Thu, 20 Mar 2014 15:01:24 +0800
Subject: [PATCH 3771/5965] meson8: clock: choose another oscillator when
 syspll is not locked in some defect chip

We have two oscillators in System PLL, one for high frequency and one for low. The boundary between two shifts a lot for different wafer lot. So in the code, if system PLL can't lock in the boundary area, change to another oscillator.
---
 arch/arm/mach-meson8/clock.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-meson8/clock.c b/arch/arm/mach-meson8/clock.c
index 8c1459a4c230..b7ba93a02565 100755
--- a/arch/arm/mach-meson8/clock.c
+++ b/arch/arm/mach-meson8/clock.c
@@ -1185,6 +1185,9 @@ SETPLL:
 					aml_set_reg32_bits(P_HHI_SYS_PLL_CNTL4, 0, 10, 1);
 					printk(KERN_ERR"  HHI_SYS_PLL_CNTL4: %08x\n", 
 						aml_read_reg32(P_HHI_SYS_PLL_CNTL4));
+				}else{
+					latency.b.afc_dsel_bp_in = !latency.b.afc_dsel_bp_in;
+					printk(KERN_ERR"  INV afc_dsel_bp_in, new latency=%08x\n",latency.d32);
 				}
 				printk(KERN_ERR"  Try again!\n");
 				goto SETPLL;
-- 
2.19.0

